src/NLog/Targets/Wrappers/AsyncRequestQueue-T.cs

```diff
@@ -71,15 +71,22 @@ public AsyncRequestQueue(int requestLimit, AsyncTargetWrapperOverflowAction over
         /// </summary>
         public int RequestCount
         {
-            get { return this.logEventInfoQueue.Count; }
+            get
+            {
+                lock (this)
+                {
+                    return this.logEventInfoQueue.Count;
+                }
+            }
         }
 
         /// <summary>
         /// Enqueues another item. If the queue is overflown the appropriate
         /// action is taken as specified by <see cref="OnOverflow"/>.
         /// </summary>
         /// <param name="logEventInfo">The log event info.</param>
-        public void Enqueue(AsyncLogEventInfo logEventInfo)
+        /// <returns>Queue was empty before enqueue</returns>
+        public bool Enqueue(AsyncLogEventInfo logEventInfo)
```

Comment:
This is a breaking change. I think we need another Enqueue method next to this one.